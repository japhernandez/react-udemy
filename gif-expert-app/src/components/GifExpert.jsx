import React, { useState } from "react";
import AddCategory from "./AddCategory";
import ListCategory from "./ListCategory";

const GifExpert = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <div className="container">
      <h2>Git Expert App</h2>
      <br />
      <AddCategory setCategories={setCategories} />
      <br />
      <hr />
      <h2>Category</h2>
      <ol>
        {categories.map((category) => (
          <ListCategory category={category} key={category} />
        ))}
      </ol>
    </div>
  );
};

export default GifExpert;
