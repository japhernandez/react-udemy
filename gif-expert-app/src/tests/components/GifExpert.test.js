import React from "react";
import { shallow } from "enzyme";
import GifExpert from "../../components/GifExpert";
import GifItem from "../../components/GifItem";

describe("Component <GitExpert />", () => {
  beforeEach(() => {});
  it("should must show component correctly", () => {
    const wrapper = shallow(<GifExpert />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should show a list of categories", async () => {
    const categories = ["One Punch", "Dragon ball"];
    const wrapper = shallow(<GifExpert defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("ListCategory").length).toBe(categories.length);
  });
});
