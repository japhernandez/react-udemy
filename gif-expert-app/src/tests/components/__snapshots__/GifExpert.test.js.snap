// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <GitExpert /> should must show component correctly 1`] = `
<div
  className="container"
>
  <h2>
    Git Expert App
  </h2>
  <br />
  <AddCategory
    setCategories={[Function]}
  />
  <br />
  <hr />
  <h2>
    Category
  </h2>
  <ol />
</div>
`;

exports[`Component <GitExpert /> should show a list of categories 1`] = `
<div
  className="container"
>
  <h2>
    Git Expert App
  </h2>
  <br />
  <AddCategory
    setCategories={[Function]}
  />
  <br />
  <hr />
  <h2>
    Category
  </h2>
  <ol>
    <ListCategory
      category="One Punch"
      key="One Punch"
    />
    <ListCategory
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</div>
`;
